### ISSUE TITLE
Enhance jump lists to replace notification area icons in certain situations

### ISSUE THREAD

**Issue by jaigak** [microsoft/WindowsAppSDK#711]
# Proposal: Enhance jump lists to replace notification area icons in certain situations
Most of the notifications icons are used just to show static app options. For these kinds of apps, jumplists are much better suited however today jumplists have some limitations such as item click would activate the foreground app instead of a background task.

## Summary
With a notification icon, the app needs to run in the background all the time. If the notification icon does more than just show a few options, that's definitely fine however why do you need the app to run all the time to show options which are accessed only when the user clicks on it? Jumplists can provide more functionality such as a toggle menu item, a checkbox menu item and subflyouts for replacing such menu only notification icons. Sometimes the app may need to  modify the options in the background based on the app state, so an event could be added that would fire when jumplist is opened and can be handled by a background process of the app to modify the menu items if needed.

## Rationale

* Improves Windows experience
* More flexibility for developers using jumplists

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to activate a background task when a jumplist item is clicked as an option | Must |
| This proposal will allow developers to add interactive toggle and check box menu items | Must |
| This proposal will allow developers to detect when the jumplist is opened and perform specific actions such as modifying the menu item list based on app state if required through a background process | Must |

## Important Notes
Since the app won't be running to register for the jumplist event, it could be a specific kind of app activation.

**Comment by riverar** [comment]
Am not sold on this. I'd like to see the rationale section filled out more. It is not clear how this improves the Windows experience or is more flexible than simply providing an icon that can be attached to a simple context menu or a completely owner drawn window.

**Comment by Jaiganeshkumaran** [comment]
The point is it's not flexible than a notification icon but it that jumplist should be more flexible than what it supports currently

**Comment by riverar** [comment]
> Most of the notifications icons are used just to show static app options.
Looking at the icons in my notification area, most of the actions require the app to be running to do anything useful (e.g. show settings, create note, send feedback, connect to server, show window, etc.). So I disagree with this assertion. Do you have some apps in mind?

It might also be helpful if you call out what's missing from the current API surfaces?
* https://docs.microsoft.com/en-us/dotnet/api/system.windows.shell.jumplist
* https://docs.microsoft.com/en-us/uwp/api/Windows.UI.StartScreen.JumpList

**Comment by mveril** [comment]
I agree with you on one point and 
- Respecting the lifecycle of a modern application is essential and this is why background tasks are key points

But I disagree with you on two points
- The system tray area has always been a good place to find background activities (Cloud synchronization, Antivirus etc) and useful information (Internet access, battery etc). This has never been the role of the taskbar apps which represent quick access to favorite apps and also foreground apps and I don't want to put apps there just to control background tasks.
- Another useful feature of this area is to be able to create a flyout to allow quick access to the application. and this is not possible with your proposal. This type of flyout could look like what we have with the [contact panel](https://docs.microsoft.com/windows/uwp/contacts-and-calendar/my-people-support#running-in-the-contact-panel)

**Comment by mrlacey** [comment]
Considering these two points:

> Most of the notifications icons are used just to show static app options.

and 

> With a notification icon, the app needs to run in the background all the time. 

I'd like to propose an additional consideration.

The fact that notification area icons are so small is one of the appealing things about them. I've known companies create notification area icons because it makes functionality easy to find and doesn't take up a lot of space.

While I'm in favor of improving the options available in the jumplists for apps in the taskbar, it would also be good if:

- it was possible to create notification area entries without needing an app running in the background.

and/or

- it was possible to create a mix of different size taskbar icon buttons. (No I haven't considered all the UX implications of this, but I don't want a bigger icon taking up more space at the bottom of the screen, because it's moved from the notification area to the app list part of the taskbar.)

**Comment by Jaiganeshkumaran** [comment]
>  I've known companies create notification area icons because it makes functionality easy to find and doesn't take up a lot of space.

Jumplists in fact don't even need an icon in the taskbar. With Windows 10, they are shown in the context menu on the Start menu tiles as well as the all apps list

**Comment by Jaiganeshkumaran** [comment]
> Looking at the icons in my notification area, most of the actions require the app to be running to do anything useful (e.g. show settings, create note, send feedback, connect to server, show window, etc.). So I disagree with this assertion.

That's true however the options itself are mostly static except a few apps. Instead of running the app always, an app provided background task can be activated when options are clicked or an event that fires to the background task when opened to modify the menu items based on app's current state.
